astropy/time/tests/test_basic.py

```diff
@@ -2883,3 +2883,12 @@ def test_timedelta_empty_quantity():
 
     with pytest.raises(ValueError, match="only quantities with time units"):
         TimeDelta([] * u.m)
+
+
+def test_immutable_location():
+    # see https://github.com/astropy/astropy/issues/16061
+    loc = EarthLocation("149d33m00.5s", "-30d18m46.385s", 236.87 * u.m)
+    t = Time("2024-02-19")
+
+    with pytest.raises(AttributeError):
+        t.location = loc
```

Comment:
Can you also add a test where the time is initialized with a `location` and one tries to change it afterwards?

Reply:
I've parametrised the test to this effect !